hkmc2/shared/src/main/scala/hkmc2/codegen/Lifter.scala

```diff
@@ -233,18 +233,51 @@ class Lifter(handlerPaths: Opt[HandlerPaths])(using State, Raise):
 
     val fresh = FreshInt()
 
-    val varsMap: Map[Local, TermSymbol] = cap.map: s =>
+    val varsMap = cap.map: s =>
         val id = fresh.make
-        s -> TermSymbol(syntax.ParamBind, S(clsSym), Tree.Ident(s.nme + id + "$"))
+        val nme = s.nme + id + "$"
+        val varSym = VarSymbol(Tree.Ident(nme))
+        val fldSym = BlockMemberSymbol(nme, Nil)
+        val fldDef = TermDefinition(
```

Comment:
So what's the intuitive reason we need all that additional stuff now in the lifter?

Reply:
Ah yes, that makes sense.

Indeed, we wanted to go back to the original intent (implemented in the previous compiler) that `class Foo(x)` does _not_ introduce a public field but `class Foo(val x)` does. We also added the alternative `data class Foo(x)` which implies that field. Also, now, ctor parameters are really just (VarSymbol) parameters, cleanly distinguished from field definitions, which solves a problem we had with super calls.